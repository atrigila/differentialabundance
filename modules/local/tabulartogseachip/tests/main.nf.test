nextflow_process {

    name "Test Process TABULAR_TO_GSEA_CHIP"
    script "../main.nf"
    process "TABULAR_TO_GSEA_CHIP"
    tag "modules"
    tag "modules_nfcore"
    tag "tabular_to_gsea_chip"

    test("test_tabular_to_gsea_chip") {

        when {
            process {
                """
                input[0] = Channel.fromPath("https://raw.githubusercontent.com/nf-core/test-datasets/refs/heads/differentialabundance/modules_testdata/Mus_musculus.anno.feature_metadata.tsv")
                input[1] = ["gene_id", "gene_name"]
                """
            }
        }

        then {
            assertAll (
                { assert process.success },
                { assert snapshot(process.out).match() },
            )
        }
    }

    test("test_tabular_to_gsea_chip - stub") {

        options "-stub"

        when {
            process {
                """
                input[0] = Channel.fromPath("https://raw.githubusercontent.com/nf-core/test-datasets/refs/heads/differentialabundance/modules_testdata/Mus_musculus.anno.feature_metadata.tsv")
                input[1] = ["gene_id", "gene_name"]
                """
            }
        }

        then {
            assertAll (
                { assert process.success },
                { assert snapshot(process.out).match() },
            )
        }
    }
}
