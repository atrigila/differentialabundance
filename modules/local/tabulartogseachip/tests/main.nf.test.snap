{
    "test_tabular_to_gsea_chip - stub": {
        "content": [
            {
                "0": [
                    "stub.chip:md5,d41d8cd98f00b204e9800998ecf8427e"
                ],
                "1": [
                    "versions.yml:md5,a422b1a01e86be433a25d1776d497f5a"
                ],
                "chip": [
                    "stub.chip:md5,d41d8cd98f00b204e9800998ecf8427e"
                ],
                "versions": [
                    "versions.yml:md5,a422b1a01e86be433a25d1776d497f5a"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.10.2"
        },
        "timestamp": "2024-12-06T21:03:06.781994447"
    },
    "test_tabular_to_gsea_chip": {
        "content": [
            {
                "0": [
                    "Mus_musculus.anno.feature_metadata.chip:md5,5abf60bb982bb2ff5b8b3e37f14a21e4"
                ],
                "1": [
                    "versions.yml:md5,a422b1a01e86be433a25d1776d497f5a"
                ],
                "chip": [
                    "Mus_musculus.anno.feature_metadata.chip:md5,5abf60bb982bb2ff5b8b3e37f14a21e4"
                ],
                "versions": [
                    "versions.yml:md5,a422b1a01e86be433a25d1776d497f5a"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.10.2"
        },
        "timestamp": "2024-12-06T21:02:58.608468636"
    }
}